allensdk/brain_observatory/behavior/data_objects/metadata/ophys_experiment_metadata/average_container_depth.py

```diff
@@ -0,0 +1,55 @@
+from pynwb import NWBFile
+
+from allensdk.core import DataObject, JsonReadableInterface, LimsReadableInterface, NwbReadableInterface  # NOQA
+from allensdk.internal.api import PostgresQueryMixin
+
+
+class AverageContainerDepth(
+    DataObject,
+    LimsReadableInterface,
+    NwbReadableInterface,
+    JsonReadableInterface,
+):
+    def __init__(self, average_container_depth: int):
+        super().__init__(
+            name="average_container_depth", value=average_container_depth
+        )
+
+    @classmethod
+    def from_lims(
+        cls, ophys_experiment_id: int, lims_db: PostgresQueryMixin
+    ) -> "AverageContainerDepth":
+        query_container_id = """
+            SELECT visual_behavior_experiment_container_id
+            FROM ophys_experiments_visual_behavior_experiment_containers
+            WHERE ophys_experiment_id = {}
+        """.format(
+            ophys_experiment_id
+        )
+
+        container_id = lims_db.fetchone(query_container_id, strict=True)
+
+        query_depths = """
+            SELECT id.depth
+            FROM ophys_experiments_visual_behavior_experiment_containers ec
+            JOIN ophys_experiments oe ON oe.id = ec.ophys_experiment_id
+            JOIN ophys_sessions os ON oe.ophys_session_id = os.id
+            LEFT JOIN imaging_depths id ON id.id = oe.imaging_depth_id
+            WHERE ec.visual_behavior_experiment_container_id = {};
+        """.format(
+            container_id
+        )
+
+        imaging_depths = lims_db.fetchall(query_depths)
+        average_container_depth = round(sum(imaging_depths) /
+                                        len(imaging_depths))
+        return cls(average_container_depth=average_container_depth)
+
+    @classmethod
+    def from_json(cls, dict_repr: dict) -> "AverageContainerDepth":
+        return cls(average_container_depth=dict_repr["targeted_imaging_depth"])
```

Comment:
We're not currently recording this value in the json. Ideas:
a) Calculate this by reading in all experiment depths for the container and taking the average
b) Calculate this and store it in the json. For VBO, we used a LIMS strategy to produce the json.

Reply:
For a), the data that's included is within the files specified in the dict_repr dictionary right? I looked through this data in the test data and it wasn't clear if other experiments from the same container were in these files. 
For b), what scripts are used to generate these jsons?

Also, the dictionary key for the imaging_depth object is ["targeted_depth"](https://github.com/AllenInstitute/AllenSDK/blob/f69f0649428346a11461472f374205916756ff4c/allensdk/brain_observatory/behavior/data_objects/metadata/ophys_experiment_metadata/imaging_depth.py#L29)

From this ticket https://github.com/AllenInstitute/AllenSDK/issues/2545, I thought the point of it is that `imaging_depth` isn't the targeted depth but rather the measured ones that can fluctuate and hence we wanted to create this additional `average_container_depth` to represent the target_depth. But this dict_repr key `targeted_depth` seems to suggest otherwise?

Reply:
> For a), the data that's included is within the files specified in the dict_repr dictionary right? I looked through this data in the test data and it wasn't clear if other experiments from the same container were in these files.
The `dict_repr` contains metadata for a single experiment. So you would have to figure out a way to read the `dict_repr` for all experiments within the container. We don't currently have a `Container` abstraction to help with this.

> For b), what scripts are used to generate these jsons?
http://stash.corp.alleninstitute.org/projects/TECH/repos/lims/browse/app/strategies/behavior_ophys_write_nwb_strategy.rb

We have tried to move away from using the LIMS strategies, since none of us are ruby developers. 
For VBN ecephys sessions, we wrote python code to produce the json.
For VBN behavior only sessions, we avoided producing the json all together. 
We can talk about this further.

> Also, the dictionary key for the imaging_depth object is ["targeted_depth"](https://github.com/AllenInstitute/AllenSDK/blob/f69f0649428346a11461472f374205916756ff4c/allensdk/brain_observatory/behavior/data_objects/metadata/ophys_experiment_metadata/imaging_depth.py#L29)
This is most likely a bad naming. I think it is the actually measured depth.

Reply:
from_json will be removed in this coming ticket. 
https://app.zenhub.com/workspaces/allensdk-10-5c17f74db59cfb36f158db8c/issues/alleninstitute/allensdk/2607